codemagic.yaml

```diff
@@ -0,0 +1,99 @@
+definitions:
+  configure_environment: &configure_environment
+    instance_type: mac_mini_m1
+    max_build_duration: 60
+    cache:
+      cache_paths:
+        - $HOME/.gradle/caches
+  scripts:
+    - &detekt_on_template_compose
+      name: Run detekt on template-compose
+      working_directory: ./template-compose
+      script: ./gradlew detekt
+    - &detekt_on_template_xml
+      name: Run detekt on template-xml
+      working_directory: ./template-xml
+      script: ./gradlew detekt
+    - &unit_test_on_template_compose
+      name: Run unit tests on template-compose
+      working_directory: ./template-compose
+      script: ./gradlew koverMergedReport
+    - &unit_test_on_template_xml
+      name: Run unit tests on template-xml
+      working_directory: ./template-xml
+      script: ./gradlew koverMergedReport
+  artifacts:
+    - &artifacts_template_compose template-compose/build/reports/kover/merged/
+    - &artifacts_template_xml template-xml/build/reports/kover/merged/
+workflows:
+  unit-test-on-pr-for-template-compose:
+    name: Unit test on PR for template-compose
+    <<: *configure_environment
+    triggering:
+      events:
+        - pull_request
+      cancel_previous_builds: true
+    scripts:
+      - *detekt_on_template_compose
+      - *unit_test_on_template_compose
+    artifacts:
+      - *artifacts_template_compose
+
+  unit-test-on-pr-for-template-xml:
+    name: Unit test on PR for template-xml
+    <<: *configure_environment
+    triggering:
+      events:
+        - pull_request
+      cancel_previous_builds: true
+    scripts:
+      - *detekt_on_template_xml
+      - *unit_test_on_template_xml
+    artifacts:
+      - *artifacts_template_xml
+
+  unit-test-on-push-for-template-compose:
+    name: Unit test on push for template-compose
+    <<: *configure_environment
+    triggering:
+      events:
+        - push
+      branch_patterns:
+        - pattern: '*'
```

Comment:
@lydiasama Just curious, should we set `included:false` for this general pattern? 🤔

Reply:
@doannimble This step is for running the unit test on the feature branch when pushing. It means when you push to update your PR, it will run the CI again. If we set the `include: false`, it means to not run when we push on any branch, right? 😂 If that's your intention, so I think we should delete this step instead. 😂

Reply:
@doannimble Hmmm, I found another way to do it [here](https://docs.codemagic.io/yaml-running-builds/starting-builds-automatically/#working-with-pull-requests). Let me update.

Reply:
@lydiasama Yeah, I think we can remove this line 😄

Reply:
@lydiasama Nice 👍